src/gf.c

```diff
@@ -988,13 +996,15 @@ JL_DLLEXPORT int jl_isa_compileable_sig(
     // supertype of any other method signatures. so far we are conservative
     // and the types we find should be bigger.
     if (definition->isva) {
-        unsigned nspec_min = nargs + 1; // min number of non-vararg values before vararg
-        unsigned nspec_max = INT32_MAX; // max number of non-vararg values before vararg
+        unsigned nspec_min = nargs + 1; // min number of arg values (including tail vararg)
+        unsigned nspec_max = INT32_MAX; // max number of arg values (including tail vararg)
         jl_methtable_t *mt = jl_method_table_for(decl);
         jl_methtable_t *kwmt = mt == jl_kwcall_mt ? jl_kwmethod_table_for(decl) : mt;
         if ((jl_value_t*)mt != jl_nothing) {
             // try to refine estimate of min and max
-            if (kwmt != NULL && kwmt != jl_type_type_mt && kwmt != jl_nonfunction_mt && kwmt != jl_kwcall_mt)
+            if (definition->max_varargs != UINT8_MAX)
+                nspec_max = nspec_min = nargs + definition->max_varargs;
```

Comment:
Why do we need to set `nspec_min` too here?

Reply:
I believe the idea is that we only want to return true for "compilation-modified signatures" (i.e. signatures that were changed/widened by `jl_compilation_sig` for the purposes of specialization)

If `tt` does not need the Varargs parameter after it was expanded `max_varargs` times, then `jl_compilation_sig` returnes no result because `tt` can be used as the compilation signature directly. In that case, this function would return false.

@vtjnash please correct me if I mixed up any details here